### PR TITLE
Add support for labels to ttir analysis

### PR DESCRIPTION
**Pull request by oulgen**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #119599
* #119859
* #119838
* __->__ #119836
* #119835
* #119834

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: test/dynamo/test_triton_kernels.py [hunk 1]
```diff
@@ -1222,6 +1222,40 @@ def add_4_times_kernel(
...
+        def kernel_with_label(
+            in_ptr0,
+            in_ptr1,
+            out_ptr,
+            n_elements,
+            BLOCK_SIZE: "tl.constexpr",
+        ):
+            pid = tl.program_id(axis=0)
+            if pid > 1:
+                return
```
Comment (aakhundov) [inline test/dynamo/test_triton_kernels.py:1237]: Should we add another test where we do different things in `if` and `else` blocks with a data-dependent condition like this?

Comment (oulgen) [reply test/dynamo/test_triton_kernels.py:1237]: I already have a test for that which turns into a `scf.if` not labels

Comment (aakhundov) [reply test/dynamo/test_triton_kernels.py:1237]: Interesting, so labels are only used on `if ...: return`? or also `if` can have a body and this is still generated as label?

**Review by aakhundov — APPROVED** [review]

File: test/dynamo/test_triton_kernels.py (hunk 1 above)
Comment (oulgen) [reply test/dynamo/test_triton_kernels.py:1237]: Depends on what happens in the body, is it dynamic or not

**Pull request closed without merging**